pokemonred_puffer/environment.py

```diff
@@ -431,7 +435,7 @@ def render(self):
 
         return game_pixels_render
 
-    def _get_obs(self):
+    def _get_screen_obs(self):
```

Comment:
Maybe this function should be combined with render or render should be renamed to `_get_screen_obs`

Reply:
render()  is also used in `add_video_frame()`? 

```
    def add_video_frame(self):
        self.full_frame_writer.add_image(self.render()[:, :, 0])
        self.model_frame_writer.add_image(self.render()[:, :, 0])
```

The reason I separated it was to make _get_obs() in the custom env simpler, like this -- https://github.com/kywch/poke-trainer/blob/template/environment.py#L37

```
    def _get_obs(self):
        return {
            "screen": self._get_screen_obs(),
        }
```

Reply:
Oh if that's the case then it's possible the visited mask should be moved into `_get_screen_obs`. I haven't used the video recording in a long time and am unsure if it still works. Something for me to test in the future.

Reply:
It seems render() is returning game_pixels_render + visited_mask
```
        game_pixels_render = np.concatenate([game_pixels_render, visited_mask], axis=-1)

        return game_pixels_render
```

It looks like render() would be good target to refactor next.